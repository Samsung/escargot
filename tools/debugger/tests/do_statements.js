/*
 * Copyright (c) 2020-present Samsung Electronics Co., Ltd
 *
 *  This library is free software; you can redistribute it and/or
 *  modify it under the terms of the GNU Lesser General Public
 *  License as published by the Free Software Foundation; either
 *  version 2.1 of the License, or (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *  Lesser General Public License for more details.
 *
 *  You should have received a copy of the GNU Lesser General Public
 *  License along with this library; if not, write to the Free Software
 *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
 *  USA
 */

function f () {
  let i = 2;
  if (i < 10) {
    if (i > 1) {
      i++;
    }
  }

  if (i > 10) {
    i++;
  } else {
    i++;
  }

  while (i > 0) {
    i--;
  }

  do {
    i++
  } while (i < 2);

  try {
    throw i;
  } catch (e) {
    i--;
  }

  for (let i = 0; i < 1; i++) {
    i++;
  }

  for (let i of [1]) {
    i++;
  }

  for (let i in { a: 8 }) {
    i++;
  }
}

f();
