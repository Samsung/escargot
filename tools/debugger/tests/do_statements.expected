Connecting to: localhost:6501
Connection created!!!
Stopped at tools/debugger/tests/do_statements.js:61
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:21 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:22 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:23 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:24 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:28 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:31 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:34 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:35 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:35 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:35 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:35 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:38 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:39 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:39 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:43 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:45 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:48 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:49 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:52 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:53 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:56 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:57 (in f() at line:20, col:1)
(escargot-debugger) s
Stopped at tools/debugger/tests/do_statements.js:59 (in f() at line:20, col:1)
(escargot-debugger) s
Connection closed.
